### PR TITLE
docs: post-v1.0.0 documentation consolidation + fresh-eyes drift sweep

### LINKED ISSUE #295: [DOCUMENTATION]: Reconcile SKILL.md producer summary after #294 (RF pre-generation on last round)
Follow-up to #294 (perf+fix: NaN hardening, fp16 default ON / bf16 off, regularization arc, index audit, RF pre-gen, SHAP overlap).

## Summary of changes that require documentation

PR #294 rewrote `docs/TRAINING_PIPELINE.md`'s "Round data: memmaps + background producer" section to reflect a new producer behavior: the `RoundDataProducer` now **also pre-generates the RF dataset while the *last* beta-VAE round trains** (the request carries `dir_name="rf"` overrides plus the `num_training_rounds + 1` seed sentinel, and `train_random_forest` awaits the result and owns shutdown from there). Previously the producer only generated round *k+1* while round *k* trained, and always shut down at the end of the round loop.

`.claude/skills/aetherscan-repo-context/SKILL.md` still says the producer "generates round k+1 while round k trains" — factually incomplete for the last-round case now that the canonical training-pipeline doc has been updated. This is contract (b) drift from PR #294.

The rest of the canonical-doc changes in #294 — the numerics-flag verdicts (`training.round_array_dtype` default → `"float16"`, `beta_vae.mixed_precision` staying `False`), the new `beta_vae.regularization_active` flag, the DB schema bump (v5 → v7 with `training_stats.is_finite`, resilient flush, and the `injection_stats` / `latent_snapshots` index sweep), the SHAP-plot overlap via `shap_pool`, and the GIF sweep 24 → 18 combos (`n_neighbors=30` dropped) — restate specific facts (defaults, schema versions, counts, flag verdicts) that neither `CLAUDE.md` nor `SKILL.md` currently mentions, so no other drift is triggered.

`CLAUDE.md` is unaffected (its always-on rules don't restate any of the facts touched by this PR).

`src/aetherscan/cli.py` was NOT touched by this PR, so contract (a) (README `## CLI Reference` regeneration) is not triggered.

## Specific files that need updates

- `.claude/skills/aetherscan-repo-context/SKILL.md` — one line (line 158, the Architecture Patterns bullet on the background data producer).

### SKILL.md edits (apply manually — .claude/ is read-only in CI)

**Edit 1** — expand the producer bullet to reflect the last-round RF pre-generation. Anchor on the current text so the edit applies unambiguously.

Old (line 158, one long bullet):

```
- **Background data producer** — `RoundDataProducer` (spawn-started process with its own worker pool) generates round k+1 while round k trains; registered with ResourceManager as a `ManagedProcess`. `CUDA_VISIBLE_DEVICES` is blanked so the producer tree never initializes CUDA; logging crosses the spawn boundary via a `QueueListener` relay.
```

New:

```
- **Background data producer** — `RoundDataProducer` (spawn-started process with its own worker pool) generates round k+1 while round k trains, and during the **last** round instead pre-generates the RF dataset (into `round_data/{tag}/rf/`, with the `num_training_rounds + 1` seed sentinel — byte-identical to `rf_train`'s in-process fallback), which `train_random_forest` awaits and owns shutdown for; registered with ResourceManager as a `ManagedProcess`. `CUDA_VISIBLE_DEVICES` is blanked so the producer tree never initializes CUDA; logging crosses the spawn boundary via a `QueueListener` relay.
```

No other SKILL.md lines need changes for PR #294. In particular:
- The `db/db.py` file-descriptor line (line 130) is generic ("schema migration, supersede semantics") and does not restate a specific version, so v5 → v7 does not drift it.
- The `shap_parallel.py` (line 126) and `latent_gif.py` (line 128) lines are generic file descriptors and do not restate SHAP or GIF-sweep specifics.
- The Configuration & CLI section (lines 63–98) does not restate any specific `training.round_array_dtype` / `beta_vae.mixed_precision` / `beta_vae.regularization_active` default.

## @claude follow-up instructions

Please open a PR linking to this issue with the following contents:

1. **Do NOT attempt to modify `.claude/skills/aetherscan-repo-context/SKILL.md`** — `.claude/` is read-only in CI and the change will fail.

2. **No `CLAUDE.md` changes are needed for this issue** — its always-on rules don't restate any of the facts changed by #294. Leave `CLAUDE.md` untouched.

3. **In the PR body, copy the `### SKILL.md edits (apply manually — .claude/ is read-only in CI)` section above verbatim**, followed by these step-by-step apply instructions for the code owner:

   ```
   # Check out the PR branch locally
   gh pr checkout <this PR number>

   # Open .claude/skills/aetherscan-repo-context/SKILL.md and apply Edit 1 above
   # (replace the single background-data-producer bullet on line 158 with the New text).

   # Commit and push back to the branch
   git add .claude/skills/aetherscan-repo-context/SKILL.md
   git commit -m "docs(skill): expand producer bullet to cover last-round RF pre-generation (#294)"
   git push
   ```

4. **Tag @zachtheyek in the PR body** (per `CODEOWNERS`, they own `.claude/`) to perform the local apply above.

The PR should otherwise be empty of file changes — its sole purpose is to carry the verbatim SKILL.md edits and the human-apply instructions into a reviewable, merge-gated location.

### LINKED ISSUE #304: [DOCUMENTATION]: Add pip-install usage instructions once v1.0.0 is released
**Gated on the v1.0.0 release** (do not start before the release PR lands — same gating pattern as #183). Maintainer-directed (interactive session, 2026-07-31).

**Problem.** The docs only cover the two source-tree install paths — the NGC container (`./utils/run_container.sh ...`) and the conda env (`PYTHONPATH=src ...`). Once the v1.0.0 CD publishes to PyPI, `pip install aetherscan` becomes the third (and easiest) path, and it is currently undocumented beyond the release runbook's smoke check.

**What to add (after v1.0.0 exists):**
- README: extend the install matrix + Usage Examples with the pip path — `pip install aetherscan==1.0.0`, then `python -m aetherscan.main inference ...` (no `PYTHONPATH=src`, no repo checkout). Note the console-script surface (`aetherscan-dashboard`) also arrives via pip.
- Call out the version-coupled weights behavior that makes pip the happy path: an installed release resolves the HF revision to `v{__version__}`, so a bare no-artifact inference download pulls exactly the v1.0.0 weights (see docs/INFERENCE_PIPELINE.md "Artifact resolution" — link rather than duplicate).
- State the environment expectations for pip users (GPU/TF 2.17 ABI constraints, `.env`/`AETHERSCAN_*` path variables, Slack/HF tokens optional) and when to prefer the container instead (Blackwell hosts).
- docs/GPU_RUNTIME_GUIDE.md / CONFIG_AND_CLI.md: one-line pointers to the pip path where the install matrix is referenced.

**Acceptance:** a fresh-machine reader can go from `pip install` to a working bare inference run (downloading v1.0.0 weights) using only the docs; the release-runbook smoke (`pip install aetherscan==1.0.0`) doubles as the verification.

### LINKED ISSUE #311: [DOCUMENTATION]: Note new PRAGMA temp_store=MEMORY in docs/DATABASE.md after #309
Follow-up from merged PR #309 (perf round-3 audit).

## Summary of the change that needs documenting

Commit `c861a7f` in #309 added a second per-connection SQLite pragma to `Database._get_connection()` in [`src/aetherscan/db/db.py`](../blob/master/src/aetherscan/db/db.py) — `PRAGMA temp_store=MEMORY`, alongside the pre-existing `PRAGMA synchronous=NORMAL` (from #277).

`docs/DATABASE.md` (line 43) is the only doc that catalogs the per-connection pragmas — it explicitly calls out `synchronous=NORMAL` with its rationale (the WAL-fsync trade-off). After #309 there are now **two** such pragmas set together on every connection, but only one is mentioned in the doc. That leaves a completeness gap in the doc that lists them: the WAL/durability sentence still reads as if `synchronous=NORMAL` is the only pragma set on `_get_connection()`.

No other canonical doc is affected by #309 — every other change in that PR is a byte-identical internal optimization (Stage-A stats memoization, `prepare_latent_features` optional `dtype` param, `prepare_distributed_train_dataset` optional `concat_only` param, first-cadence-only bandpass-envelope retention, frequency-width geometry validation, non-winner variant-forest freeing) at a level the docs don't currently discuss, and #309 did not touch `src/aetherscan/cli.py` so the CLI Reference is unaffected.

## Files that need updates

- **`docs/DATABASE.md`** — the paragraph ending at line 46 that mentions `PRAGMA synchronous=NORMAL`. Add a short follow-on sentence naming the new pragma and its rationale, in the same style as the existing one, and mirroring the code comment in [`src/aetherscan/db/db.py`](../blob/master/src/aetherscan/db/db.py) added by commit `c861a7f`.

## Suggested change

Extend the paragraph at `docs/DATABASE.md:43-46` with a second per-connection pragma sentence. Something like:

> Connections also set `PRAGMA temp_store=MEMORY` so the large `GROUP BY` / `ORDER BY` / window-function sorts on the read-heavy teardown/plot passes (per-round injection-stat aggregates, the decimated resource-monitor window query, whole-tag plot scans) stay in RAM instead of spilling to an on-disk temp b-tree; small result sets never spill, so it is free otherwise. Per-connection and numerics-neutral. `mmap_size` is deliberately **not** set: its benefit is largely lost under the per-query open/close model, and SQLite mmap I/O faults surface as `SIGBUS` on shared cluster nodes — not worth the marginal gain.

Exact wording is at the writer's discretion — the load-bearing points are (a) that `temp_store=MEMORY` is now set alongside `synchronous=NORMAL` on every connection, (b) *why* (the read-heavy sort passes), and (c) *why not* `mmap_size` (SIGBUS risk on shared nodes + benefit lost to per-query open/close).

## Contracts

- **Contract (a) — CLI Reference**: not triggered — #309 did not touch `src/aetherscan/cli.py`, so README's `## CLI Reference` blocks are unaffected.
- **Contract (b) — CLAUDE.md / SKILL.md reconciliation**: not triggered — #309 changed only source code, no canonical doc, and neither `CLAUDE.md` nor `.claude/skills/aetherscan-repo-context/SKILL.md` currently states any fact this PR made wrong (SKILL.md's DB reference just points at `docs/DATABASE.md`; CLAUDE.md doesn't discuss DB internals). No CLAUDE.md or SKILL.md edits are needed for this issue.

## Instructions to @claude

Please open a PR that:

1. Adds a short follow-on sentence to the existing per-connection-pragma paragraph in `docs/DATABASE.md` (currently ending around line 46) that names `PRAGMA temp_store=MEMORY`, explains *why* (large read-heavy `GROUP BY` / `ORDER BY` / window sorts on teardown/plot passes stay in RAM), and briefly notes *why not* `mmap_size` (SIGBUS risk on shared cluster nodes + benefit lost to the per-query open/close model) — mirroring the code comment added in commit `c861a7f` of #309.
2. Keeps the existing `PRAGMA synchronous=NORMAL` sentence and its WAL-fsync rationale intact.
3. Links this issue via `Closes #<this-issue-number>` in the PR body.
4. Does NOT touch `.claude/skills/aetherscan-repo-context/SKILL.md` or `CLAUDE.md` (neither is affected by this issue).
5. Does NOT modify any `.py` source or any other doc — scope is `docs/DATABASE.md` only.

### PR DESCRIPTION
**Pull request by zachtheyek**
## What

The unified post-v1.0.0 documentation pass. It **folds the four auto-generated `claude-update-docs` carrier PRs** into one reviewed PR, fills the remaining documentation gaps from the doc-issue backlog, and runs a fresh-eyes docs-vs-code drift sweep over the fully-merged master.

Closes #311, Closes #295, Closes #304.

## Folded carrier PRs (close after this merges)

The docs bot opened one PR per merge in the 5-PR cleanup batch; their content is verified and rolled in here, so they should be closed in favour of this PR:

- **#326** → `docs/RELEASE.md` packaging extras (`pytest>=9.0.3` security floor, `streamlit>=1.54`)
- **#328** → `benchmarks/README.md` + `docs/BENCHMARKING.md` bench `--data-dir` defaults
- **#331** → the #318 feature docs across 9 files (per-band plot, RF variant-selection plot, feature-count guard, candidate frequency axis)
- **#332** → `KNOWN_ISSUES.md` + `docs/MODELS.md` pre-#293 artifact-compat facts

(Their paired auto-issues #325/#327/#329/#330 close with them.)

## New documentation (the gaps the bot could not cover)

- **#311** — `docs/DATABASE.md`: `PRAGMA temp_store=MEMORY` (and why `mmap_size` is deliberately unset — SIGBUS on shared nodes).
- **#295** — `SKILL.md` producer bullet: the `RoundDataProducer` also pre-generates the RF dataset while the **last** round trains.
- **#304** — `README.md`: a PyPI pip-install path documenting the **v1.0.0 workaround** (`tf_keras~=2.17.0` + `TF_USE_LEGACY_KERAS=1`), the Blackwell-needs-the-container caveat (`CUDA_ERROR_INVALID_PTX`), and the no-CPU-mode caveat. The workaround steps are marked as removable once the v1.0.1 fix (#323) ships.
- **HF "unsafe" flag** — `SECURITY.md`: ProtectAI flags `vae_encoder.keras` because loading it deserializes the registered custom `Sampling` layer; documented as benign (accept + document).
- **Display-tag convention (#322)** — `docs/ARCHITECTURE.md`: filenames / plot titles / Slack use `{cmd}_{machine}_{datetime}` while DB rows keep `{cmd}_{datetime}`; the umap-joblib plain-tag exception; the cross-host `--load-tag` caveat. (No carrier PR covered #322 — it merged last.)

## Fresh-eyes drift sweep

A full docs-vs-code audit surfaced **23 findings; 22 are fixed here** (each verified against a concrete `file:line`). Highlights:

- `ARCHITECTURE.md`: preprocessed stamps are **ED-config-fingerprint-scoped** (`<csv_stem>_ed<hash12>`), not tag-scoped.
- `TRAINING_PIPELINE.md`: the "Disk budget" numbers were the **float32** figures — the default is `float16` (~147 GB/round, not ~295 GB).
- `MODELS.md`: SHAP feature labels are `{ON|OFF}-{pair}_dim-{d}` (from `latent_variants.variant_feature_names`), not `obs{i}_z{d}`.
- `INFERENCE_PIPELINE.md`: the viz latent pool is a **budget-fill**, not reservoir sampling (reservoir is the reference cloud).
- `DATABASE.md`: the "second-heaviest table" parenthetical named a nonexistent `signal_characteristics` (→ `injection_stats`); `Added in` columns missing v6/v3.
- `CONFIG_AND_CLI.md`: `--save-tag` is resolved in `main()` via `resolve_save_tag`, not an `apply_args_to_config` Pattern B flag.
- `BENCHMARKING.md`: the `epochs` span is a nested `stage_timer` (not `record_stage`); added the `inference.reference_cloud` span and the two schema-v7 DB-index benches.
- `GITHUB_AUTOMATION.md`: `claude-dependency-check` audits `pyproject.toml` too; `tests.yml` also has a `workflow_call` trigger; dropped a wrong `CONTRIBUTING.md` citation.
- `TESTING.md`: added `test_monitor.py`, `test_perband_report.py`, `test_seeding_determinism.py` to the trees.

**Rejected-ideas note:** HANDOFF-era "rejected ideas" (bf16-off, fp16 round-arrays, SHAP CPU-vs-GPU incl. the log-loss/depth reasons, prefetch RAM) turned out to be **already well-documented** in `TRAINING_PIPELINE.md`/`CONFIG_AND_CLI.md`/`INFERENCE_PIPELINE.md`, so this PR only adds the genuinely-missing gaps (the #293 sweep specifics, and that fp16 is round-arrays-only — stamps stay float32) plus a factual note that the RF `max_depth` is unbounded.

## Deferred (surfaced, not folded)

One sweep finding is **code-side**: the release tooling (`utils/hf_tag_release.py`, `.github/workflows/release.yml`) cites stale `docs/RELEASE.md` runbook step numbers (the runbook was renumbered). Left to a focused release-tooling PR rather than folded into a docs PR — several citations drifted and the correct values need remapping against the current runbook.

## Testing

Docs-only (18 `.md` files). All new cross-reference anchors verified to resolve; pre-commit clean.

### CODE REVIEW

**Comment by zachtheyek** [comment]
Thanks — and good catch on both blocking items; they were a real slip.

**Blocking #1 + #2 (`AGENTS.md` + `.agents/skills/.../SKILL.md`) — removed.** These were pre-existing **untracked** files in the working tree at the start of this session (a `claude`→`Codex` agent-config experiment, not part of this docs work). A `git add -A` for the docs commit swept them in, and my pre-commit `git diff --stat` didn't show them because `git diff` doesn't list untracked files — so I missed them entirely. You're right that they're also factually broken (the `@Codex` / `Codex.yml` / `.Codex` facts don't exist in this repo, and the handle one inverts the #83 footgun rule). I've `git rm --cached`'d both and amended the commit, so the PR is back to the intended **18 `.md` files** and the two files return to untracked WIP for whoever picks up the Codex config separately. Consequences of that:
- **#295 is fully closed** by the real `.claude/skills/.../SKILL.md` producer-bullet edit (the `.agents/` copy was never in scope); the half-closed divergence you flagged goes away with the file.
- The `.agents/` staleness (`InfDataHolder`, missing `candidate_figures.py`) and the "single canonical path vs 282-line duplicate" concern are real but belong to that future Codex-config PR, not this one — noting them there.

**Non-blocking #3 (`README.md:31`) — fixed.** Dropped the now-redundant trailing sentence; the lead-in ("two source-tree paths … plus the published PyPI package") carries the canonical/alternative distinction without the verbatim restatement.

**Non-blocking #4 — moot after the removal.** With `AGENTS.md` gone there's a single authoritative agent-config file again (`CLAUDE.md`), so the existing `docs/ARCHITECTURE.md:185` / `docs/GITHUB_AUTOMATION.md:35` / `docs/RUNTIME_SERVICES.md:123` pointers to it stay correct, and the `CONTRIBUTING.md` Project-Structure tree needs no `AGENTS.md`/`.agents/` entry.

Also noted for the record: the one drift the sweep found that this PR deliberately does **not** fix is the release tooling's stale `docs/RELEASE.md` step-number citations (`utils/hf_tag_release.py`, `.github/workflows/release.yml`) — that's code, not docs, and several citations need remapping against the renumbered runbook, so it's tracked for a focused follow-up rather than folded here.

@claude please take another pass.

**Comment by zachtheyek** [comment]
Thanks for the LGTM and the thorough re-verification.

Addressed the one non-blocking note in d20f885: added a caveat that a **pip install skips the two end-of-run report PNGs** — the wheel ships only `src/aetherscan`, so `utils/benchmark_report.py` / `utils/perband_report.py` aren't present and both plots log-and-skip (everything else is unaffected) — and softened the data-root wording (only `AETHERSCAN_DATA_PATH` must pre-hold inputs; the model/output roots are made on demand).

Not re-tagging, since we're at LGTM. Merging once the docs-only CI is green; the four auto-carrier PRs (#326 / #328 / #331 / #332) will be closed as folded here.

For the record, your inference about the base-provided convention was exactly right — it bit the sibling #323 fix (#340): omitting `tf_keras` from `requirements-container.txt` (the correct call) means `tests.yml` has to name it explicitly, and it didn't yet, so that PR's CI went red at collection. Fixing it there.
